tests/sentry/search/events/builder/test_metrics.py

```diff
@@ -2314,6 +2314,76 @@ def test_run_query_with_on_demand_eps(self):
             {"name": "eps", "type": "Float64"},
         ]
 
+    def test_run_query_with_on_demand_user_misery(self):
+        threshold = 300
+        field = f"user_misery({threshold})"
+        query_s = "transaction.duration:>=10"
+        spec = OnDemandMetricSpec(field=field, query=query_s)
+
+        transaction_duration_repeats = [
+            (threshold, 3),  # satisfied
+            (threshold + 100, 2),  # tolerated
+            ((threshold * 4) + 1, 10),  # frustrated
+        ]
+
+        for hour in range(0, 2):
+            for duration, repeat in transaction_duration_repeats:
+                for _ in range(0, repeat):
+                    self.store_transaction_metric(
```

Comment:
I need to figure out how to have more than one user in these stored transactions.